pkg/operator/controller/controller.go

```diff
@@ -247,6 +249,19 @@ func (r *reconciler) enforceEffectiveHighAvailability(ci *ingressv1alpha1.Cluste
 	return nil
 }
 
+// ensureClusterIngressFinalized adds ClusterIngressFinalizer to ingress
+// if it does not exist and does not contain a DeletionTimestamp.
+func (r *reconciler) ensureClusterIngressFinalized(ingress *ingressv1alpha1.ClusterIngress) error {
+	if ingress.ObjectMeta.DeletionTimestamp.IsZero() && !slice.ContainsString(ingress.Finalizers, ClusterIngressFinalizer) || ingress.Finalizers == nil {
+		ingress.Finalizers = append(ingress.Finalizers, ClusterIngressFinalizer)
+		if err := r.Client.Update(context.TODO(), ingress); err != nil {
```

Comment:
@Miciah @pravisankar did we decide our "policy" on mutations? As @Miciah previously explained we're pretty safe given the particulars of our client implementation. The policy could just be to note in GoDoc that the parameter will be modified. WDYT?

Reply:
Using controller-runtime's API, it is clear and natural to use the client as Dane is using it here—that is, we declare the `ingressv1alpha1.ClusterIngress{}` in `Reconcile` and pass around a pointer to it, and we count on `r.Client.Update` to update it for us.  How likely are we to move away from the controller-runtime client?

Reply:
Okay, I agree with you, I know you've been trying to hammer this into my head for a while... just making sure. I think we're all on the same page and that we can rely on Update and Create to "refresh" existing references we're passing around.

Reply:
It also means we're free to remove the `DeepCopy()` calls floating around, but that's just random cleanup